testplan/web_ui/testing/src/Report/BatchReportBeta/state/uiSelectors.js

```diff
@@ -0,0 +1,82 @@
+import { createSelector } from '@reduxjs/toolkit/dist/redux-toolkit.esm';
+import { css } from 'aphrodite/es';
+import _spread from 'lodash/spread';
+import _isObject from 'lodash/isObject';
+import { mkGetReportDocument } from './reportSelectors';
+import navStyles from '../../../Toolbar/navStyles';
+import { STATUS_CATEGORY } from '../../../Common/defaults';
+import { COLUMN_WIDTH } from '../../../Common/defaults';
+
+const STATUS_CATEGORY_STYLE_MAP = {
+  passed: navStyles.toolbarPassed,
+  failed: navStyles.toolbarFailed,
+  error: navStyles.toolbarFailed,
+  unstable: navStyles.toolbarUnstable,
+};
+
+export const createUISelector = (...funcs) => {
```

Comment:
"But, like any abstraction, it's not something you should do all the time, everywhere. Writing selectors means more code to understand and maintain. Don't feel like you need to write selectors for every single field of your state. "

Reply:
plus selectors created with createSelector memoize what's selected